### ISSUE TITLE
JSC doesn't support regex lookbehinds (breaks e.g. discord.js)

### ISSUE THREAD

**Issue by NullDev** [oven-sh/bun#314]
Unfortunately JSC still does not support regex lookbehinds (<https://caniuse.com/js-regexp-lookbehind>) which breaks packages like [discord.js](https://www.npmjs.com/package/discord.js). The error happens as soon as either `import` or `require` try to load a file using a lookbehind. In discord.js's case, it leads to `TypeError: Invalid regular expression: invalid group specifier name while parsing module` which is a JSC specific error:

```js
1 | 'use strict';
2 | 
3 | const EventEmitter = require('node:events');
4 | const RESTManager = require('../rest/RESTManager');
5 | const Options = require('../util/Options');
6 | const Util = require('../util/Util');
                        ^
```
```
 TypeError: Invalid regular expression: invalid group specifier name while parsing module "/home/shad0w/projects/BunTest/node_modules/discord.js/src/util/Util.js"
      at /home/shad0w/projects/BunTest/node_modules/discord.js/src/client/BaseClient.js:6:21
      at bun:wrap:1:16354
      at /home/shad0w/projects/BunTest/node_modules/discord.js/src/index.js:4:29
      at bun:wrap:1:16354
      at /home/shad0w/projects/BunTest/app.ts:4:0
```

Import: <https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/client/BaseClient.js#L7>
(probably) caused by line: <https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/util/Util.js#L109>

**Comment by cohenerickson** [comment]
I have a similar issue with discord.js but it seems to be related to a package called `@sapphire/async-queue` imported from https://github.com/discordjs/discord.js/blob/stable/src/rest/RequestHandler.js#L5

```
error: Could not resolve: "@sapphire/async-queue". Maybe you need to "bun install"?

const { AsyncQueue } = require('@sapphire/async-queue');
```

When I try to run `bun add @sapphire/async-queue` or `bun install` I get this error
```
[2.45ms] ".env"
bun install v0.1.1
 Failed to install 6 packages
[273.00ms] done
error: NotSameFileSystem installing @types/node
error: NotSameFileSystem installing @sapphire/shapeshift
error: NotSameFileSystem installing @jridgewell/resolve-uri
error: NotSameFileSystem installing @sapphire/async-queue
error: NotSameFileSystem installing @discordjs/builders
error: NotSameFileSystem installing @sapphire/async-queue
```

**Comment by Jarred-Sumner** [comment]
@cohenerickson This bug is that `bun install` isn't correctly detecting the existence of a remote filesystem

For now, you can work around it by passing `--backend=copyfile`, like so:
```bash
bun install --backend=copyfile
```

**Comment by wbourne0** [comment]
webkit ticket: https://bugs.webkit.org/show_bug.cgi?id=174931

**Comment by AlexAyoub545** [comment]
I tried running the 

```bun install --backend=copyfile```

still giving me the same issue, any other workarounds or?

**Comment by FinnRG** [comment]
@AlexAyoub545  This is a webkit issue

**Comment by AlexAyoub545** [comment]
@FinnRG Alrighty thank you

**Comment by Elior2739** [comment]
@AllAwesome497 This ticket got reported in 2017 and there isn't fix for that
What are the chances that they will fix this?

**Comment by FinnRG** [comment]
> What are the chances that they will fix this?

@Elior2739 Well, Webkit is known to have the worst compatibility of the browser engines, but in the last few months there was a lot of work in this area. Since this is a relatively important feature, chances aren't too bad that this will be fixed this year.

**Comment by ghost** [comment]
As a workaround, would Bun be open to implementing the lookbehinds, and rewriting the `RegExp` bindings and prototype?
If that is successful, then a patch could be sent upstream for WebKit.

**Comment by Milo123459** [comment]
Hey, are there any updates on this?

**Comment by ghost** [comment]
> As a workaround, would Bun be open to implementing the lookbehinds, and rewriting the `RegExp` bindings and prototype?
> If that is successful, then a patch could be sent upstream for WebKit.

(Well, a comment from myself - I neglected to consider that *regexp literals would remain broken*, and even a naïve implementation into JSC's RegExp compiler doesn't seem easy)

**Comment by Jarred-Sumner** [comment]
This issue is fixed by https://github.com/oven-sh/bun/pull/1329

**Comment by Jarred-Sumner** [comment]
This has been fixed for most cases in Bun v0.2.1. 

The two remaining TODOs:
- Tweak JSC's `RegExp` constructor to fallback automatically to `Bun.OngurumaRegExp` when it fails due to a missing feature. 
- Remove `Bun.OngurumaRegExp` once JSC lands builtin support for regex lookbehinds
